modules/fundamental/src/vulnerability/model/details/vulnerability_advisory.rs

```diff
@@ -476,6 +650,46 @@ impl VulnerabilitySbomStatus {
             }
         }
 
+        for product in products.clone() {
+            if let Some(sbom) = &product.sbom {
+                let advisory_cpe = (&product.cpe).try_into().ok();
+                let entry = VulnerabilitySbomStatus {
+                    head: SbomHead::from_entity(sbom, product.sbom_node.clone(), tx).await?,
+                    version: product
+                        .product_version
+                        .as_ref()
+                        .map(|pv| pv.version.clone()),
+                    cpe: advisory_cpe,
+                    status: vec![product.status.slug.clone()].into_iter().collect(),
+                };
+                sboms
+                    .entry(&sbom.sbom_id)
+                    .and_modify(|value| {
+                        match &value.cpe {
+                            Some(cpe) => {
+                                // This is where the matching logic for multiple advisory entries lives
+                                // At the moment we try to use the biggest range we have.
+                                // For example: if we have entries for 2.0, 2.7 and 2.13 we will choose the last one
+                                if let Some(version) = &product.cpe.version {
+                                    let version_semver = lenient_semver::parse(version);
+                                    let current_version = &cpe.version().clone().to_string();
+                                    let current_semver = lenient_semver::parse(current_version);
+
+                                    if match (&version_semver, &current_semver) {
```

Comment:
Have you considered using existing version related functions (e.g. psql funcs) so we do not introduce a 2nd code path (using **lenient_semver**) - I believe all those functions can be used directly in rust/sea_orm (or maybe even better performing in the query itself.

Reply:
I did, but I was pushing what I was able to make work at this point with pure sql. I would leave that for the future improvement, I would love to have all this incorporated in the original query

Reply:
sounds good -  might be worth inserting a TODO here.

Reply:
I'll add it, but it definitely won't be forgotten :)